monkey/monkey_island/cc/environment/test__init__.py

```diff
@@ -0,0 +1,113 @@
+import json
+import os
+from typing import Dict
+from unittest import TestCase
+from unittest.mock import patch, MagicMock
+
+from common.utils.exceptions import InvalidRegistrationCredentials
+from monkey_island.cc.environment import Environment, EnvironmentConfig, UserCreds
+
+
+def get_server_config_file_path_test_version():
+    return os.path.join(os.getcwd(), 'test_config.json')
+
+
+class TestEnvironment(TestCase):
+
+    class EnvironmentNoCredentials(Environment):
+        _credentials_required = False
+
+        def get_auth_users(self):
+            return []
+
+    class EnvironmentWithCredentials(Environment):
+        _credentials_required = True
+
+        def get_auth_users(self):
+            return []
+
+    # Username:test Password:test
+    CONFIG_WITH_CREDENTIALS = {
```

Comment:
Making tests scattered all around the codebase might've been a mistake: where should common code between tests go?

Reply:
Under common?